### ISSUE TITLE
Mouse positions wrong in iframe in Chrome with "Use worker" enabled.

### ISSUE THREAD

**Issue by MathNook1** [Scirra/Construct-3-bugs#3748]
## Problem description
When drag and drop behavior is enabled on a sprite and the project has "Use worker" enabled it works fine in an iframe on Edge but not in Chrome.  In Chrome the dragged object will often be offset far away from where the cursor is located.  Hopefully it is something that Scirra can work around or at least give everyone a heads up.

## Attach a .c3p

https://www.dropbox.com/s/w6hzlz1sejt6uad/dragdropworker.c3p?dl=0

## Steps to reproduce

1. Download from attached link or simply create a project with "Use Worker" enabled and a sprite with drag and drop behavior
2. export HTML
3. upload html to website
4. create webpage with the index.html called in an iframe
5. run webpage in Chrome where you should see the sprite offset a good distance from where the cursor is.
6. Run in Edge which works fine.

## Expected result

Expected the drag and drop to follow the cursor as it does in the editor and in Edge rather than there be an offset of the sprite from the cursor 

by various distances.

**Affected browsers/platforms:** 

Version 80.0.3987.149 (Official Build) (64-bit)

[dragDropWithWorker.zip](https://github.com/Scirra/Construct-3-bugs/files/4438502/dragDropWithWorker.zip)

**Comment by AshleyScirra** [comment]
This looks strikingly similar to issue #3743, which used custom styles on the exported page. Are you sure you're not doing the same? It's possible to break this depending on the styles you use. Please share the exported page using the iframe that you are using, too.

**Comment by MathNook1** [comment]
It could be....I'll have to look into it further but a couple of things to note:  The page will work fine in Edge but not Chrome.  If I disable "Use Worker" and export the game and run it in the same page with the same code and CSS it works fine in Chrome.  Same HTML and CSS used in lots of games (mine and from Game Distribution and Game Monitization).  Had this happen several weeks back in another game that utilized drag and drop.

**Comment by MathNook1** [comment]
Wait.  If you're talking about modifying the style.css exported from Construct 3, no it wasn't modified other than to change the background color.

**Comment by MathNook1** [comment]
Ashely, you were correct it was due to the margin setting on the Div. I narrowed it down to the issue comes up in Chrome whenever the margin is set on the Div the iframe is in.  The larger the margin the larger the offset.  I realize this is a Chrome issue and that I can avoid this behavior by disabling "Use Worker" but is there anything in Construct 3 to identify the correct location so I can have the sprite follow the mouse without offset?

**Comment by AshleyScirra** [comment]
Hmm it might still be an issue with Construct, or some kind of browser bug, since moving the iframe should not affect the contents of the iframe in any way. I'll try this out myself in a bit.

**Comment by MathNook1** [comment]
Below is the HTML I used:

<!DOCTYPE html>
<html lang="en-US">
  <meta name="description" content=" "/>
<head>
  <meta charset="utf-8">
</head>
<body>

<div style="padding;300px;margin:50px;">

	    <iframe  src="https://www.website.com/dragdrop/withworker/index.html" frameborder="0" marginheight="0" marginwidth="0" scrolling="no" ></iframe>
</div>

</body>
</html>

**Comment by MathNook1** [comment]
Attached is the html I used.
[drag-drop-with-worker-bare-html v2.zip](https://github.com/Scirra/Construct-3-bugs/files/4439346/drag-drop-with-worker-bare-html.v2.zip)

**Comment by Gildas44** [comment]
I confirm that with no marge and no worker it works for me too.

**Comment by AshleyScirra** [comment]
It turns out this is a bug in Chrome - in some cases it gives the wrong mouse positions in an iframe. I filed it with Google: [crbug.com/1068958](https://bugs.chromium.org/p/chromium/issues/detail?id=1068958)
Meanwhile I've added a workaround for the next release.

We seem to have received a few similar reports about this at around the same time, which is odd because nothing has changed on our side since r189.2 came out a month ago. Maybe it's a regression in Chrome, or somehow nobody noticed until now. Anyway the workaround means it should be working again in the next release.
